### PR TITLE
[CoreAnimation] CATextLayer.Alignment* strings should be CATextLayerAlignment enum / type. Fixes bug 59537

### PR DESCRIPTION
**Pull request by dalexsoto**
https://bugzilla.xamarin.com/show_bug.cgi?id=59537

Removes constants from `CATextLayer` (only for XAMCORE_4_0) and creates
two smart enums `CATextLayerTruncationMode` and `CATextLayerAlignmentMode`.
Also this introduces two new strong properties into CATextLayer class,
`TextTruncationMode` and `TextAlignmentMode` that takes the new enums
respectively, these new properties are meant to replace their
string counterparts `TruncationMode` and `AlignmentMode`.

### CODE REVIEW

File: src/CoreAnimation/CATextLayer.cs [hunk 1]
```diff
@@ -134,6 +134,52 @@ public void SetFont (NSFont font)
...
+				NSString.ReleaseNative (nsvalue);
+			}
+		}
+#endif // !XAMCORE_4_0
+		public CATextLayerTruncationMode TextTruncationMode {
+			get {
+				return CATextLayerTruncationModeExtensions.GetValue (Runtime.GetNSObject<NSString> (_TruncationMode));
+			}
+			set {
+				_TruncationMode = value.GetConstant ().Handle;
```
Comment (spouliot) [inline src/CoreAnimation/CATextLayer.cs]: `value.GetConstant ()` could be `null`, e.g. if the constant does not exists in the executing OS version, leading to an NRE
why not have `_TruncationMode` be an `NSString` and let it deal with `null` ?

File: src/CoreAnimation/CATextLayer.cs [hunk 2]
```diff
@@ -134,6 +134,52 @@ public void SetFont (NSFont font)
...
+			}
+		}
+
+		public CATextLayerAlignmentMode TextAlignmentMode {
+			get {
+				return CATextLayerAlignmentModeExtensions.GetValue (Runtime.GetNSObject<NSString> (_AlignmentMode));
+			}
+			set {
+				_AlignmentMode = value.GetConstant ().Handle;
```
Comment (spouliot) [inline src/CoreAnimation/CATextLayer.cs]: same

File: src/CoreAnimation/CATextLayer.cs [hunk 3]
```diff
@@ -134,6 +134,52 @@ public void SetFont (NSFont font)
...
+				var nsvalue = NSString.CreateNative (value);
+				_TruncationMode = nsvalue;
+				NSString.ReleaseNative (nsvalue);
+			}
+		}
+
+		[Advice ("Use 'TextAlignmentMode' instead.")]
+		public string AlignmentMode {
+			get {
+				return NSString.FromHandle (_AlignmentMode);
```
Comment (spouliot) [inline src/CoreAnimation/CATextLayer.cs]: you can `return (string) _AlignmentMode` if it was an `NSString`

File: src/CoreAnimation/CATextLayer.cs [hunk 4]
```diff
@@ -134,6 +134,52 @@ public void SetFont (NSFont font)
...
+			}
+			set {
+				if (value == null)
+					throw new ArgumentNullException (nameof (value));
+				var nsvalue = NSString.CreateNative (value);
+				_AlignmentMode = nsvalue;
+				NSString.ReleaseNative (nsvalue);
```
Comment (spouliot) [inline src/CoreAnimation/CATextLayer.cs]: you can do `_AlignmentMode = (NSString) value;`

**Comment by dalexsoto** [comment]
@spouliot fixed and thanks!! Great suggestion!

**Comment by dalexsoto** [comment]
@spouliot done 👍

File: src/CoreAnimation/CATextLayer.cs [hunk 5]
```diff
@@ -134,6 +134,28 @@ public void SetFont (NSFont font)
 				}
 			}
 		}
+#if !XAMCORE_4_0
+		[Advice ("Use 'TextTruncationMode' instead.")]
```
Comment (spouliot) [inline src/CoreAnimation/CATextLayer.cs]: please use `[Obsolete]` instead (it's inside XAMCORE_4_0)

**Pull request merged**